### PR TITLE
using ObjectId for Id in User Collection.

### PR DESCRIPTION
**Pull request by hman-pro**
I think this leads to breaking change! 
I follow to hear what's your idea...
#29

### CODE REVIEW

**Comment by tugberkugurlu** [comment]
thanks for this 🎉

File: src/AspNetCore.Identity.MongoDB/MongoUserStore.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ public MongoUserStore(IMongoDatabase database, ILoggerFactory loggerFactory, str
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            await _usersCollection.InsertOneAsync(user, cancellationToken).ConfigureAwait(false);
+            await _usersCollection.InsertOneAsync(user, null, cancellationToken).ConfigureAwait(false);
```
Comment (tugberkugurlu) [inline src/AspNetCore.Identity.MongoDB/MongoUserStore.cs:74]: what does this `null` mean here?

File: src/AspNetCore.Identity.MongoDB/MongoIdentityUser.cs [hunk 2]
```diff
@@ -33,7 +33,7 @@ public MongoIdentityUser(string userName)
                 throw new ArgumentNullException(nameof(userName));
             }
 
-            Id = GenerateId(userName);
+            // Id = GenerateId(userName); // Id will generate by mongodb driver
```
Comment (tugberkugurlu) [inline src/AspNetCore.Identity.MongoDB/MongoIdentityUser.cs]: we can delete this comment and `GenerateId` impl. I believe `GenerateId` is not used anymore?

Comment (hman-pro) [reply src/AspNetCore.Identity.MongoDB/MongoIdentityUser.cs]: yep, alright.

File: src/AspNetCore.Identity.MongoDB/MongoUserStore.cs (hunk 1 above)
Comment (hman-pro) [reply src/AspNetCore.Identity.MongoDB/MongoUserStore.cs:74]: the previews method was obsoleted with this message => Use the new overload of InsertOneAsync with an InsertOneOptions parameter instead.

**Comment by hman-pro** [comment]
@tugberkugurlu  let me know if there is any problem in this PR :)

File: src/AspNetCore.Identity.MongoDB/MongoUserStore.cs (hunk 1 above)
Comment (tugberkugurlu) [reply src/AspNetCore.Identity.MongoDB/MongoUserStore.cs:74]: that makes sense. However, I will change the usage to below one which seems a bit cleaner:

```csharp
await _usersCollection.InsertOneAsync(user, cancellationToken: cancellationToken).ConfigureAwait(false);
```

File: src/AspNetCore.Identity.MongoDB/MongoIdentityUser.cs (hunk 2 above)
Comment (tugberkugurlu) [reply src/AspNetCore.Identity.MongoDB/MongoIdentityUser.cs]: OTH, we need to give it an within the ctor: `Id = ObjectId.GenerateNewId().ToString();` I will do that, too.

**Comment by tugberkugurlu** [comment]
I rebased and merged this in, @SorenZ your commits are in: b5f736508bfae107d32a340acd7dc624dad9f338 and a4d570376047f422e1a9b7f942cd892f36820183. Thanks!

**Pull request closed without merging**